bom/runtime/pom.xml

```diff
@@ -2177,6 +2177,11 @@
                 <artifactId>apache-client</artifactId>
                 <version>${awssdk.version}</version>
             </dependency>
+            <dependency>
+                <groupId>software.amazon.awssdk</groupId>
+                <artifactId>url-connection-client</artifactId>
+                <version>${awssdk.version}</version>
+            </dependency>
```

Comment:
What annoys me a bit is that you get all the dependencies with the extension. I wonder if it's a wise choice and we shouldn't be more careful about that especially in the lambda use case.

That being said, I'm not totally sure we can find a nice way to do that properly but IMHO it's worth a try.

If we can get to the point of the dependencies being optional and we can throw a warning or an error if something is configured for a "backend" and the correct dependencies are not in the classpath, that would be OK IMHO.

I don't think it's blocking for this PR tbh so I think we could merge this and see how we can improve things later.

Reply:
I like the idea and in fact tried it, check the latest commits. In general:
1. I made client deps as optional
2. the deployment processor checks if required service exists on the classpath, if not throws deploymentException
3. the application should explicitly add required dependency to the classpath

Do you think it make sense ? Once you verify the idea, I'll update docs & quickstart